packages/babel-plugin-transform-spread/src/index.js

```diff
@@ -18,6 +18,28 @@ export default function({ types: t }, options) {
     return false;
   }
 
+  function isConsumableArrayHelper(node) {
```

Comment:
Actually this is not a good way of checking if something is a `consumableArray` helper. Helpers come in different forms - consider simple example with "inline" helpers:
```js
"use strict";

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

[].concat(_toConsumableArray(smth));
```

Im not sure about this transform's logic, but I see that `scope.toArray` is actually inserting the helper, is all info needed for determining `isNewArray` test's result available then (when `scope.toArray` happens)? Maybe this could be moved into `scope.toArray`? If not we need a better test for `isConsumableArrayHelper` but recently I had the same problem and I didnt found an answer (yet) whats the best way to get that info without creating the helper

Reply:
Yeah, that makes sense. I figured there would be some concerns. Let me think about how this can be restructured.

Reply:
check if this can be restructured without actually knowing anything about `toConsumableArray` - if it got inserted by this plugin, the check can probably happen some time before inserting it, on input nodes or something

Reply:
Yep, I think I actually have a much simpler way of doing this after looking at what `toArray` (& related code) does.